### PR TITLE
Feature/fix unclear error message

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
wrap all errors in a `WorkflowError` including a business error message

### CODE REVIEW

File: src/dataProtector/grantAccess.ts [hunk 1]
```diff
@@ -56,36 +57,45 @@ export const grantAccess = async ({
...
-    const datasetorder = await iexec.order.signDatasetorder(
-      datasetorderTemplate
-    );
-    await iexec.order.publishDatasetorder(datasetorder);
-    return formatGrantedAccess(datasetorder);
-  } catch (error) {
-    throw new Error(`Failed to grant access: ${error.message}`);
-  }
+  await iexec.order.publishDatasetorder(datasetorder).catch((e) => {
+    throw new WorkflowError('Failed to publish access', e);
```
Comment (abbesBenayache) [inline src/dataProtector/grantAccess.ts:98]:
```suggestion
 throw new WorkflowError('Failed to publish data access', e);

Comment (abbesBenayache) [reply src/dataProtector/grantAccess.ts:98]: Ou bien un autre message, à mon avis le message n'est pas tres claire :)

File: src/dataProtector/protectDataObservable.ts [hunk 2]
```diff
@@ -135,16 +135,22 @@ export const protectDataObservable = ({
           name: vName,
```
Comment (abbesBenayache) [inline src/dataProtector/protectDataObservable.ts:135]: const ownerAddress = await signer.getAddress();

we can catch error : 

const ownerAddress = await signer.getAddress().catch((e: Error) => {
          throw new WorkflowError('Failed to retrieve the owner's address', e);
        });

File: tests/e2e/IExecDataProtector/grantAccess.test.ts [hunk 3]
```diff
@@ -128,17 +128,19 @@ describe('dataProtector.grantAccess()', () => {
...
       dataProtector.grantAccess({ ...input, authorizedApp: nonTeeAppAddress })
     ).rejects.toThrow(
-      Error(
-        'Failed to grant access: App does not use a supported TEE framework'
+      new WorkflowError(
+        'Failed to detect the app TEE framework',
+        Error('App does not use a supported TEE framework')
       )
     );
   });
```
Comment (abbesBenayache) [inline tests/e2e/IExecDataProtector/grantAccess.test.ts:146]: We can create a ticket in the next sprint to add unit tests, mocking, and testing all error-catching scenarios.

**Review by abbesBenayache — APPROVED** [review]

File: src/dataProtector/grantAccess.ts [hunk 4]
```diff
@@ -56,36 +57,45 @@ export const grantAccess = async ({
...
       requesterrestrict: vAuthorizedUser,
       datasetprice: vPricePerAccess,
       volume: vNumberOfAccess,
       tag,
+    })
+    .then((datasetorderTemplate) =>
+      iexec.order.signDatasetorder(datasetorderTemplate)
+    )
+    .catch((e) => {
+      throw new WorkflowError('Failed to sign access', e);
```
Comment (PierreJeanjacquot) [inline src/dataProtector/grantAccess.ts:95]:
```suggestion
      throw new WorkflowError('Failed to sign data access', e);
```

File: src/dataProtector/protectDataObservable.ts (hunk 2 above)
Comment (PierreJeanjacquot) [reply src/dataProtector/protectDataObservable.ts:135]: This is not expected to happen in the nominal workflow.
If this happens there is nothing the end user can do, so we will just throw an "unexpected error"

File: tests/e2e/IExecDataProtector/grantAccess.test.ts (hunk 3 above)
Comment (PierreJeanjacquot) [reply tests/e2e/IExecDataProtector/grantAccess.test.ts:146]: yes I drafted a unit test implementation that works but I'm not completely happy with it in the way things are mocked or stubbed. you can have a look on it https://github.com/iExecBlockchainComputing/dataprotector-sdk/pull/68

**Pull request merged**